pkg/vault/contracts/Vault.sol

```diff
@@ -1185,43 +1187,54 @@ contract Vault is IVaultMain, VaultCommon, Proxy {
             _bufferTokenBalances[wrappedToken] = bufferBalances;
         } else {
             // The buffer does not have enough liquidity to facilitate the wrap without making an external call.
-            // We wrap the user's tokens via an external call and additionally rebalance the buffer if it has a
-            // surplus of underlying tokens.
+            // We wrap the user's tokens via an external call and additionally rebalance the buffer if it has an
+            // imbalance of underlying tokens.
 
-            // Gets the amount of underlying to wrap in order to rebalance the buffer.
-            uint256 bufferUnderlyingSurplus = bufferBalances.getBufferUnderlyingSurplus(wrappedToken);
-            uint256 bufferWrappedSurplus;
+            // Gets the imbalance of underlying in the buffer. If negative, the buffer has more wrapped than underlying.
+            int256 bufferUnderlyingImbalance = bufferBalances.getBufferUnderlyingImbalance(wrappedToken);
+            int256 bufferWrappedImbalance = bufferBalances.getBufferWrappedImbalance(wrappedToken);
```

Comment:
So, this is a long thread of thought:
* When we are wrapping ExactOut, we need to make sure that we have enough amountOutWrapped to give to the user at the end. 
* In the previous approach, I was using `vaultWrappedDeltaHint = previewDeposit(amountInUnderlying + bufferUnderlyingImbalance)`, which works but is not 100% accurate. I can't ensure that I'll have enough amountOutWrapped and that the buffer will be balanced, because I'm not using the ExactOut amounts to calculate the mint argument. 
* Calculating both imbalances does not increase gas costs and reduces the bytecode (especially in the unwrap part, where I don't need to approve anything and saved one external call).

Reply:
After some more thought, I reverted and did as you said. Calculating the buffer imbalance already calls an external call, so it won't make a difference, and the 80 bytes of saving was not due to this change, but related to the typo in the `if` clause of the _wrapWithBuffer function that I described below.